x-pack/plugins/encrypted_saved_objects/server/plugin.ts

```diff
@@ -92,17 +88,21 @@ export class Plugin {
         service.registerType(typeRegistration),
       __legacyCompat: { registerLegacyAPI: (legacyAPI: LegacyAPI) => (this.legacyAPI = legacyAPI) },
       usingEphemeralEncryptionKey,
+      startWithHiddenTypes: (includedHiddenTypes: string[]) =>
```

Comment:
I haven't looked through all the encrypted saved objects plugin's code, so I'll leave this up to the security team, but this API is different from what I would have expected because we're exposing two almost identical API's from setup and start.

Perhaps when there's a hidden type registration through `registerType` we could automatically include this hidden type into the underlying repository or alternatively there could be a setup API like `includeHiddenTypes: (includedHiddenTypes: string[])` which just sets a `includedHiddenTypes` class property which is then read when creating the `getDecryptedAsInternalUser` api is created.

Reply:
Yeah, I was a little thrown by the fact that the Start api *is* the client, rather than returning an api for creating a client.
Given that it felt like maybe the Setup api should provide a way of returning a version of the start api with the additional hidden type, otherwise all encryptedSavedObjects clients in Kibana will, by default, have access to the hidden types... which seems weird to me. 🤔 

But yeah, I'm open to changing this to whatever makes sense to the Security team obviously.

Would love to hear @legrego 's thoughts

Reply:
Yes you're right, my suggestion will open up hidden types to all consumers which isn't what we want.

Reply:
> Yeah, I was a little thrown by the fact that the Start api is the client, rather than returning an api for creating a client.

@azasypkin do you recall if there's a reason we expose the client as the start contract, rather than exposing a function to create the client? It seems that if we can expose the function to create, then we can use that function to optionally construct a client which includes access to hidden types.

Reply:
I'm happy to expose a suitable api on the Start contract if you'd like.
We'd need to retain the existing API as removing would be a breaking change, but at least we'd have a single api for including hidden types and just a straight up regular client, and perhaps we remove the old api in 8.0?

Reply:
> I'm happy to expose a suitable api on the Start contract if you'd like.

That would be great, thanks! Let's wait to hear from @azasypkin in case I'm overlooking something thouh.

> We'd need to retain the existing API as removing would be a breaking change

We don't offer any guarantees for plugin API stability _yet_, so introducing a breaking change is perfectly OK to do from my perspective. Encrypted Saved Objects are still pretty new, and it's a relatively esoteric feature, so I don't expect that there is broad (or any) adoption of this outside of Kibana itself.

Reply:
> @azasypkin do you recall if there's a reason we expose the client as the start contract, rather than exposing a function to create the client?

No particular reason, just nobody needed anything more advanced at that time.

> 

> I'm happy to expose a suitable api on the Start contract if you'd like.
> That would be great, thanks! Let's wait to hear from @azasypkin in case I'm overlooking something thouh.

:100: 

> We don't offer any guarantees for plugin API stability yet, so introducing a breaking change is perfectly OK to do from my perspective. Encrypted Saved Objects are still pretty new, and it's a relatively esoteric feature, so I don't expect that there is broad (or any) adoption of this outside of Kibana itself.

Definitely agree, let's not create a tech debt when we have a chance to avoid that.

Reply:
Cool, it shall be done :)

Reply:
I've now exposed a `getClient` api and changed the few places that are using ESO to use this api instead.

Please let me know if this is what you had in mind. :)